middleware/fields/file-upload.js

```diff
@@ -0,0 +1,61 @@
+/**
+ * File Upload.
+ */
+
+const { FormFileUpload } = wp.components;
+const { __ } = wp.i18n;
+const { mediaUpload } = wp.utils;
+
+export default function fileUpload( props, config, attributeKey ) {
+	const buttonText = config.buttonText ? config.buttonText : __( 'Upload' );
+	const fileType = config.fileType ? config.fileType : 'image';
+
+	const setMedia = ( file ) => {
+		const newAttributes = {};
+		newAttributes[ attributeKey ] = file;
+		props.setAttributes( newAttributes );
+	};
+
+	const defaultAttributes = {
+
+		onChange( event ) {
+			mediaUpload( event.target.files, setMedia, fileType );
+		},
+
+		fileType: fileType,
+
+		accept: "image/*",
```

Comment:
`image/*` shouldn't be the default, we should probably allow few common file types by default. And if somebody wants to allow certain file types they should be able to pass an array.